imap_processing/hi/l1b/hi_l1b.py

```diff
@@ -42,9 +53,67 @@ def hi_l1b(l1a_cdf_path: Path):
             comment="#",
             converters={"mnemonic": str.lower},
         )
+
         l1b_dataset.attrs.update(hi_hk_l1b_global_attrs.output())
-        return l1b_dataset
-    elif packet_enum in (HIAPID.H45_SCI_DE, HIAPID.H90_SCI_DE):
+    elif logical_source_parts[-1].endswith("de"):
+        l1b_dataset = annotate_direct_events(l1a_dataset)
+    else:
         raise NotImplementedError(
-            f"L1B processing not implemented for file: {l1a_cdf_path.name}"
+            f"No Hi L1B processing defined for file type: "
+            f"{l1a_dataset.attrs["Logical_source"]}"
+        )
+    return l1b_dataset
+
+
+def annotate_direct_events(l1a_dataset):
+    """
+    Perform Hi L1B processing on direct event data.
+
+    Parameters
+    ----------
+    l1a_dataset: xarray.Dataset
+        L1A direct event data.
+
+    Returns
+    -------
+    xarray.Dataset
+        L1B direct event data.
+    """
+    n_epoch = l1a_dataset["epoch"].size
+    new_data_vars = dict()
+    for var in [
+        "coincidence_type",
+        "esa_step",
+        "delta_t_ab",
+        "delta_t_ac1",
+        "delta_t_bc1",
+        "delta_t_c1c2",
+        "spin_phase",
+        "hae_latitude",
+        "hae_longitude",
+        "quality_flag",
+        "nominal_bin",
+    ]:
+        attrs = CDF_MANAGER.variable_attributes[f"hi_de_{var}"]
```

Comment:
I think this is where @maxinelasp was concerned with returning mutable instances of things, where the `pop` below will be impacting future lookups. I don't remember where everything stands, but worth hearing from her on this use.
```suggestion
        attrs = CDF_MANAGER.get_variable_attributes(f"hi_de_{var}")
```

Reply:
I will also use `copy()` here to avoid unwanted mutation.